src/Native/System.Native/pal_interfaceaddresses.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include "pal_maphardwaretype.h"
+#include "pal_types.h"
+
+struct LinkLayerAddressInfo
+{
+    uint32_t InterfaceIndex; // The index of the interface to which this address belongs.
+    uint8_t AddressBytes[8]; // A pointer to the bytes containing the address.
+    uint8_t NumAddressBytes; // The number of bytes actually stored in the address.
+    uint8_t __padding;
+    NetworkInterfaceType HardwareType;
+};
+
+struct IpAddressInfo
+{
+    uint32_t InterfaceIndex;
+    uint8_t AddressBytes[16];
+    uint8_t NumAddressBytes;
+    uint8_t __padding[3];
+};
+
+typedef void (*IPv4AddressFound)(const char* interfaceName, IpAddressInfo* addressInfo, IpAddressInfo* netMaskInfo);
```

Comment:
Nit: if these typedefs aren't used in externally-visible function signatures, you could sink them into the source file.